### PR TITLE
chore: github oidc least privilege

### PR DESCRIPTION
**Pull request by insignias**
Issue #, if available:

Description of changes:
1. github oidc least privilege

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: workbench-core/repo-scripts/github-oidc/README.md [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+⚠️ $\textcolor{red}{\text{Experimental}}$ ⚠️ : Not for use in any critical, production, or otherwise important deployments
+
+## Description
+This package creates the roles in an AWS account that can be assumed from a github workflow. The permissions given to these roles enable the workflow to perform deployments in the specified AWS account. 2 CDK stacks are created as part of this package:
+1. `OIDCProviderStack`: This stack deploys the identity provider for GitHub
+2. `<GitHub-Org>-GitHubOIDCStack`: This stack deploys the role-to-assume which will be used in our github workflows to deploy and perform integration/e2e tests
+
+## Constructs
+This package defines 2 constructs:
+1. [MafGithubOidcRolePolicy](./src/constructs/maf-github-oidc-role-policy.ts)
```
Comment (rsmayda) [inline workbench-core/repo-scripts/github-oidc/README.md:12]: Do we need this one anymore?

File: workbench-core/repo-scripts/github-oidc/package.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "@aws/workbench-core-github-oidc",
```
Comment (rsmayda) [inline workbench-core/repo-scripts/github-oidc/package.json:2]: Should this be a package? in the SWB repo? -- we can sync offline -- curious to know what SWB SDEs think

File: workbench-core/repo-scripts/github-oidc/README.md (hunk 1 above)
Comment (insignias) [reply workbench-core/repo-scripts/github-oidc/README.md:12]: Yes we do !

File: workbench-core/repo-scripts/github-oidc/package.json (hunk 2 above)
Comment (insignias) [reply workbench-core/repo-scripts/github-oidc/package.json:2]: We are not going to publish it. but are you thinking from security point of view ?

File: common/config/rush/browser-approved-packages.json
Comment (DaveMadsen) [inline common/config/rush/browser-approved-packages.json:1]: did GitHub OIDC really require all of these changes or is there a lot here that can go into a separate PR?

File: rush.json [hunk 3]
```diff
@@ -91,12 +91,6 @@
       "reviewCategory": "production",
       "shouldPublish": true
     },
-    {
```
Comment (DaveMadsen) [inline rush.json:94]: this probably should have been a separate change that went in before the OIDC change.

File: common/config/rush/browser-approved-packages.json
Comment (insignias) [reply common/config/rush/browser-approved-packages.json:1]: This PR just creates the `github-oidc` package and all the changes are related to that. I also updated the [reviewCategory](https://github.com/aws-solutions/solution-spark-on-aws/pull/986/files#diff-7cf530b920932d4075d799c0aa86e3757def01fb357fcb27ebc090868ceb62ddR139) to tools which might have caused this big change

File: workbench-core/repo-scripts/github-oidc/NOTICE [hunk 4]
```diff
@@ -0,0 +1,1160 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (DaveMadsen) [inline workbench-core/repo-scripts/github-oidc/NOTICE:756]: this should be reported to the owners of aws/constructs.

File: workbench-core/repo-scripts/github-oidc/README.md (hunk 1 above)
Comment (DaveMadsen) [inline workbench-core/repo-scripts/github-oidc/README.md:12]: maf --> wbc (workbench-core)

File: workbench-core/repo-scripts/github-oidc/README.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+2. [SwbGithubOidcRolePolicy](./src/constructs/swb-github-oidc-role-policy.ts)
+
+Use these constructs to update the permissions according to the requirement.
+
+### MAF Example Package Instructions
```
Comment (DaveMadsen) [inline workbench-core/repo-scripts/github-oidc/README.md:17]: If you're not going to use "workbench-core" consistent with the rest of this project, you at lest need to define "MAF" here before using the acronym.

File: workbench-core/repo-scripts/github-oidc/src/cdk-bootstrap-policies/maf/maf-hosting-cdk-bootstrap-policy-us-east-1.json
Comment (DaveMadsen) [inline workbench-core/repo-scripts/github-oidc/src/cdk-bootstrap-policies/maf/maf-hosting-cdk-bootstrap-policy-us-east-1.json:1]: I noticed you have `us-east-1` both hardcoded and in the file name. Should the file name include `policy-aws-us-east-1` as you are hardcoding the `aws` partition as well?

File: workbench-core/repo-scripts/github-oidc/NOTICE (hunk 4 above)
Comment (insignias) [reply workbench-core/repo-scripts/github-oidc/NOTICE:756]: Done: https://github.com/aws/constructs/issues/1648

**Review by DaveMadsen — DISMISSED** [review]
It looks like there's really two separate changes here:
- remove example-ui package
- github OIDC
I would prefer these were separated into different PRs.

Couple of other comments, mainly the use of `maf` in contexts where we've used `wbc` previously, and using `MAF` in documentation without defining it.

File: workbench-core/repo-scripts/github-oidc/src/cdk-bootstrap-policies/maf/maf-hosting-cdk-bootstrap-policy-us-east-1.json
Comment (insignias) [reply workbench-core/repo-scripts/github-oidc/src/cdk-bootstrap-policies/maf/maf-hosting-cdk-bootstrap-policy-us-east-1.json:1]: Good call out, I will update the file name. Regarding the region, I created separate documents that are required to setup our integration test environments. I have called it out in the [README](https://github.com/aws-solutions/solution-spark-on-aws/pull/986/files#diff-b8b28413da00e46ec144a2a355fb529748ce7bca1bd980bf6683b04ff13d3144R21)

**Comment by insignias** [comment]
> It looks like there's really two separate changes here:
> 
> * remove example-ui package
> * github OIDC
>   I would prefer these were separated into different PRs.
> 
> Couple of other comments, mainly the use of `maf` in contexts where we've used `wbc` previously, and using `MAF` in documentation without defining it.

Thanks @DaveMadsen I will push a fix

File: rush.json (hunk 3 above)
Comment (DaveMadsen) [reply rush.json:94]: Also, if this package is going away, does this need to go to? https://github.com/aws-solutions/solution-spark-on-aws/blob/49f18e511a91f97bb0498650794123d8d424dded/.github/workflows/build-and-test.yml#L30
